Enable github action CI by switching to vertica db

blackeyepanda opened the pull request:
initial attempt to switch the postgres db in github actions to vertica db s.t. we can run tests

blackeyepanda committed:
initial update to git action CI to use vertica db

blackeyepanda committed:
try fix ci issue by adding boostrap

blackeyepanda committed:
Update ci.yml

jeremyprime commented on .github/workflows/ci.yml:
```diff
@@ -1,32 +1,53 @@
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
Should these be master instead of main?

blackeyepanda replied:
Yep

jeremyprime commented on .github/workflows/ci.yml:
```diff
@@ -1,32 +1,53 @@
+          docker build -f docker-vertica/Dockerfile.centos.7_9.x --build-arg VERTICA_PACKAGE=vertica-ce.latest.rpm -t jbfavre/vertica docker-vertica
+          docker images
+          docker run -d -p 5433:5433 jbfavre/vertica
+          sleep 60
```
Why not just use the official [vertica-ce Docker image](https://hub.docker.com/r/vertica/vertica-ce)?

```
docker run -d -p 5433:5433 vertica/vertica-ce:11.1.1-0
```

blackeyepanda replied:
I just copied what we did in vertica-python and vertica-sql-go, I am OK with either way for that as long as it is consistent across all the open source drivers

jeremyprime replied:
I think either the [Vertica CE](https://hub.docker.com/r/vertica/vertica-ce) (Ubuntu 18) or [Vertica Kubernetes](https://hub.docker.com/r/vertica/vertica-k8s) (CentOS 7.9) image would be the way to go since they are both still actively managed and updated.  We use the Vertica Kubernetes image for testing the Spark connector, although either one is fine since we are simply accessing the Vertica server and not the OS.

That docker-vertica repo is quite old and hasn't been updated in years, and it requires additional dependencies and steps when a simple docker run would suffice.

sitingren replied:
@blackeyepanda You can use the [Vertica CE](https://hub.docker.com/r/vertica/vertica-ce) (Ubuntu 18) or [Vertica Kubernetes](https://hub.docker.com/r/vertica/vertica-k8s) (CentOS 7.9) image. vertica-python and vertica-sql-go just haven't done the switch to these latest images.

williammatherjones replied:
Looks like ci.yml has been updated to use Vertica CE.  Has this conversation been resolved?

blackeyepanda replied:
yep, I will use Vertica CE for this pr, then we can change the vertica docker image python and go client use in the future, I will create PR for them lately.

blackeyepanda committed:
Merge branch 'vertica:master' into master

blackeyepanda committed:
update main branch in ci to master

blackeyepanda committed:
Merge branch 'vertica:master' into master

blackeyepanda committed:
separate into 3 sub tests

vertica-nodejs test suites is not passing yet, so running other tests suites separate first

blackeyepanda committed:
reduce to only one node version for now

blackeyepanda committed:
Merge branch 'vertica:master' into master

blackeyepanda committed:
update to use vertica-ce docker image

blackeyepanda committed:
more updates to ci

- seems no need to wait 60s
- try more vertica instance with different node version

blackeyepanda committed:
adding missing -d to detach the docker container

blackeyepanda committed:
update CI

- keep test running on error
- currently more than one Vertica docker instance will cause conflict ports, only use one for now

blackeyepanda committed:
Merge branch 'vertica:master' into master

williammatherjones commented on .github/workflows/ci.yml:
```diff
@@ -1,32 +1,64 @@
+           --name vertica_ce \
+           vertica/vertica-ce
+
+      - name: test-v-connection-string
```
The CI is currently running tests for v-connection-string, v-pool and v-protocol.    Yang told me we should omit the tests for vertica-nodejs until they are passing.    Should we merge this PR now, or wait until we have passing tests for vertica-nodejs and include that package test?

DMickens replied:
I'm all for merging this now if it's ready for those packages. It would have caught the v-connection-string unit test failure that we discovered yesterday.

blackeyepanda replied:
I have added the vertica-nodejs package as well, as expected, there are a lot failures there, we can working on fixing them after the ci merged.

DMickens replied:
adding the vertica-nodejs package will be useful even if there are a lot of failures. It's just another opportunity to catch a regression

jeremyprime commented on .github/workflows/ci.yml:
```diff
@@ -1,32 +1,64 @@
+      - name: Setup Vertica
+        run: |          
+          docker pull vertica/vertica-ce:12.0.0-0
+          docker run -d -p 5433:5433 -p 5444:5444 \
```
This takes some time to startup and configure the DB.  Do we need to add a sleep after docker run?  This may be related to the current test failures.

If you manually perform this docker run locally you can see what an appropriate sleep might be (30s, 60s, etc, of course it may need to be different in GitHub).

jeremyprime replied:
If the tests still fail when accessing localhost, another option is to use the service name as the hostname (`V_HOST=vertica_ce` in this case).

blackeyepanda replied:
yep, I think the reason the v-protocol package didn't pass before is due to Vertica not up totally, I have added a vsql probe to make sure Vertica startup when these tests running

jeremyprime commented on .github/workflows/ci.yml:
```diff
@@ -1,32 +1,64 @@
+        if: always()
+        run: |
+          cd packages/v-connection-string
+          V_HOST=localhost, V_PORT=5433, V_USER=dbadmin, V_DATABASE=VMart yarn test
```
If possible would be nice to export these env vars in a prior step so they are not repeated.

blackeyepanda committed:
Merge branch 'vertica:master' into master

blackeyepanda committed:
try to make ci available for check-in

1. currently using only node v14 for testsing, will investigate mult-version tests later
2. run all the test packages for all even though it may failed

blackeyepanda committed:
set environment variables

williammatherjones commented:
Let's log any outstanding issues in Jira and merge this!   The only condition is that failures do not prevent PRs from being merged until the tests stabilize.

DMickens commented on .github/workflows/ci.yml:
```diff
@@ -1,32 +1,84 @@
+        # let's make it a little bit simple for now
+        # current target version will be 14.
+        # TODO: investigate the multipe version matrix with single Vertica instance
+        node: ['14']
```
We should probably log a jira to expand this once things settle down with testing

blackeyepanda replied:
I have created one for that

Commit:
adding vsql probe to make Vertica startup
